command/agent/cache/lease_cache.go

```diff
@@ -159,6 +159,13 @@ func (c *LeaseCache) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 		return nil, err
 	}
 
+	// If this is a 3xx or if the returned response does not contain JSON payload,
+	// we skip caching
+	if (resp.Response.StatusCode > 300 && resp.Response.StatusCode < 400) ||
```

Comment:
We might want to change this to `>=300`.

Reply:
Yea we can definitely do that. Another thing to note, and that we should adjust as well, is that responses that result in errors returned by `c.proxier.Send()` (https://github.com/hashicorp/vault/pull/6353/files#diff-703e1234dc09e66de48735d23f23732eR157) will end up getting the actual response being wrapped and a 500 being returned instead of the original response status code:
```
› curl -v 127.0.0.1:8007/v1/foo/bar
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8007 (#0)
> GET /v1/foo/bar HTTP/1.1
> Host: 127.0.0.1:8007
> User-Agent: curl/7.54.0
> Accept: */*
> 
< HTTP/1.1 500 Internal Server Error
< Content-Type: application/json
< Date: Wed, 06 Mar 2019 03:20:16 GMT
< Content-Length: 170
< 
{"errors":["failed to get the response: Error making API request.\n\nURL: GET http://127.0.0.1:8200/v1/foo/bar\nCode: 404. Errors:\n\n* no handler for route 'foo/bar'"]}
* Connection #0 to host 127.0.0.1 left intact
```